api/src/test/java/org/pmiops/workbench/api/UserMetricsControllerTest.java

```diff
@@ -261,6 +282,27 @@ public void testGetUserRecentResource() {
     assertEquals(recentResources.get(1).getCohort().getName(), "Cohort Name");
   }
 
+  @Test
+  public void testGetUserRecentResources_missingWorkspace() {
+    resource1.setWorkspaceId(9999); // missing workspace
+    when(mockUserRecentResourceService.findAllResourcesByUser(user.getUserId()))
+        .thenReturn(ImmutableList.of(resource1, resource2));
+    RecentResourceResponse recentResources =
+        userMetricsController.getUserRecentResources().getBody();
+    assertEquals(1, recentResources.size());
```

Comment:
nit: I'm a fan of `AssertJ` for more readable assertions.

`assertThat(recentResources).hasSize(1)`

Reply:
another nit: I don't think its necessarily a bad thing but I prefer testing minimally by only asserting what would fail.

In this case, this test would have failed with a 404 on the `getUserRecentResources()` call so I would be content with stopping there.

Reply:
Re: assertion libraries, please see [this ticket](RecentResourceResponse recentResources =
        userMetricsController.getUserRecentResources().getBody()) about unifying them. It's much more important that we settle on one of them than that we pick the best one. This file is using JUnit assertions, so I just went with what we had, but I agree with you that AssertJ and Truth are much better.

Reply:
I'm not following this comment: 
_In this case, this test would have failed with a 404 on the getUserRecentResources() call so I would be content with stopping there._

The whole intention of the PR was to have it not do a 404 here but return those resources whose workspaces are actually reachable.